var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -215,6 +216,9 @@ def configure_args(self):
         ldflags.append(hdf5_hl.libs.search_flags)
         if hdf5_hl.satisfies("~shared"):
             libs.append(hdf5_hl.libs.link_flags)
+            h5_pc_cmd = ["pkg-config", "hdf5", "--static", "--libs"]
+            h5_flags = subprocess.check_output(h5_pc_cmd, encoding="utf8").strip()
```

Comment:
The idiomatic way of doing exactly this is to make sure that the executable is available at the build time when needed (i.e. `depends_on("pkgconfig", type="build", when="^hdf5~static")`)
```suggestion
            pkgconfig = which("pkg-config")
            h5_flags = pkgconfig("hdf5", "--static", "--libs", output=str)
```
Also, the result is later appended to `libs`, so it makes sense to use `--libs-only-l` instead of just `--libs` and then to make a second call with `--libs-only-L` to append the output to `ldflags`.